src/dataflow/source/kafka.rs

```diff
@@ -41,6 +41,30 @@ lazy_static! {
     .unwrap();
 }
 
+// There is other stuff in librdkafka messages, e.g. headers.
+// But this struct only contains what we actually use, to avoid unnecessary cloning.
+struct MessageParts {
+    payload: Option<Vec<u8>>,
+    partition: i32,
+    offset: i64,
+    key: Option<Vec<u8>>,
+}
+
+trait BorrowedMessageExt {
+    fn detach_parts(&self) -> MessageParts;
+}
```

Comment:
Do you think this needs a trait? Would it be enough to have a `From<BorrowedMessage<'a>>` implementation? Not really a problem, but just trying to keep things simple!

Reply:
I could do that. I wanted the name to specifically be `detach_parts` so that people with an idea of what the `BorowedMessage::detach` function does have an idea what is going on.

Reply:
Gotcha. A free function like `fn detach_parts(msg: &BorrowedMessage<'a>) -> MessageParts` probably also works! But `MessageParts::from(msg)` probably looks clear enough at the call site? Anyhow, not a big deal; up to you!

Reply:
I'll just change it to `MessageParts::from`. As you say, not really a big deal